mailpoet/lib/Captcha/ReCaptchaRenderer.php

```diff
@@ -0,0 +1,39 @@
+<?php declare(strict_types = 1);
+
+namespace MailPoet\Captcha;
+
+use MailPoet\Settings\SettingsController;
+use MailPoet\Subscription\Captcha\CaptchaConstants;
+use MailPoet\WP\Functions as WPFunctions;
+
+class ReCaptchaRenderer {
+
+  /** @var SettingsController */
+  private $settings;
+
+  /** @var WPFunctions */
+  private $wp;
+
+  public function __construct(
+    SettingsController $settings,
+    WPFunctions $wp
+  ) {
+    $this->settings = $settings;
+    $this->wp = $wp;
+  }
+
+  public function render(): string {
+    $captchaSettings = $this->settings->get('captcha');
+    $isInvisible = $captchaSettings['type'] === CaptchaConstants::TYPE_RECAPTCHA_INVISIBLE;
```

Comment:
At this point, `captcha.type` is set to either the checkbox or invisible reCAPTCHA option. Otherwise, MP's reCAPTCHA hooks won’t be added, and this function will never called.

This is a precondition for the function. I’d rather avoid bloating the logic with redundant checks.

For the built-in CAPTCHA, though, if `captcha.type` is set to `disabled`, but CAPTCHA is enabled on WP/WC forms, we should default to the built-in solution. In that case, `captcha.type` would be `nil`, right?

Reply:
> Otherwise, MP's reCAPTCHA hooks won’t be added, and this function will never called.

> This is a precondition for the function.

Seems like this is the case, indeed. Then it's good as it is. Thanks for reassuring!

> In that case, captcha.type would be nil, right?

Yes, if captcha is disabled, `type` would be `null`. My note was about a hypothetical case when the whole `captcha` setting is absent, but the `render()` method is called, but as you explained, it won't be called.